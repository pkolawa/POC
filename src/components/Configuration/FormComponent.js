import React from "react";
import T from 'prop-types';
import { withStyles } from "@material-ui/core/styles";
import FormRow from "./../FormRow";
import { TextField } from "@material-ui/core";

const styles = (theme) => ({

});


/**
 * Allowed field types
 */
export const FORM_FIELD_TYPES = {
	// form fields
    button: 'button',
    number: 'number',
	string: 'string', // text field
	select: 'select', // select
};

class FormComponent extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formFields: []
        }
    }

    static propTypes = {
        pageData: T.array.isRequired,
    };


    renderForm = pageData => {

        return pageData.map(item => {
            const { fields } = item;

            return this._renderFormRow(fields);
        })
    }

    _renderFormRow = rows => {

        return (
            <FormRow
            // header={field.header}
            // {...field.customProps}
            // classes={{}}
            >
                {rows.map(item => {
                    return this.renderSectionItems(item);
                })}
            </FormRow>
        );
    }


    renderSectionItems = (field) => {
        const { type, name, value} = field;
        
        let component;
        switch (type) {
			case FORM_FIELD_TYPES.string:
                component = this.renderTextField(name, value);
                break;
            case FORM_FIELD_TYPES.number:
                component = this.renderNumberField(name, value);
                break;
            default:
                return false
            
        }

        console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>', component);

        return (
			<React.Fragment key={field.name}>
                {component}
            </React.Fragment>
				
		);
    }

    renderTextField = (name, value) => {
        return <TextField type="text" label={name} value={value} />

    }

    renderNumberField = (name, value) => {
        return <TextField shrink type="number" label={name} value={value} />

    }





    render() {
        const { pageData } = this.props;

        return this.renderForm(pageData);

    }

    componentDidCatch(error, info) {
        console.log('>>>>>>>>>>>>..', error, info);
    }
}

export default withStyles(styles)(FormComponent);
