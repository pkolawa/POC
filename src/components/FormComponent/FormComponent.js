import React from "react";
import T from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import FormRow from "./../FormRow";
import { TextField } from "@material-ui/core";

const styles = (theme) => ({});

/**
 * Allowed field types
 */
export const FORM_FIELD_TYPES = {
  // form fields
  button: "button",
  number: "number",
  string: "string", // text field
  select: "select", // select
};

class FormComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      formFields: [],
    };
  }

  static propTypes = {
    pageData: T.array.isRequired,
  };

  renderForm = (pageData) => {
    return pageData.map((item) => this._renderFormRow(item.fields));
  };

  _renderFormRow = (rows) => {
    return (
      <FormRow
        // header={field.header}
        // {...field.customProps}
        // classes={{}}
        gridSpacing={2}
      >
        {rows.map((item,i) => {
          return <Grid item key={i}>{this.renderSectionItems(item)}</Grid>;
        })}
      </FormRow>
    );
  };

  renderSectionItems = (field) => {
    const { type, name, value } = field;

    switch (type) {
      case FORM_FIELD_TYPES.string:
        return this.renderTextField(name, value);
      case FORM_FIELD_TYPES.number:
        return this.renderNumberField(name, value);
      default:
        return false;
    }
  };

  handleOnChange = (fieldName, event) => {
    
    const newState = Object.assign({}, this.state);
    newState[fieldName] = event.target.value;
    this.setState(newState);
  }

  renderTextField = (name, value) => {
    return <TextField label={name} value={value} onChange={(e) => this.handleOnChange(name, e)} />;
  };

  renderNumberField = (name, value) => {
    return <TextField shrink type="number" label={name} value={value} />;
  };

  render() {
    return this.renderForm(this.props.pageData);
  }

  componentDidCatch(error, info) {
    console.error(">>>>>>>>>>>>..", error, info);
  }
}

export {FormComponent}

export default withStyles(styles)(FormComponent);