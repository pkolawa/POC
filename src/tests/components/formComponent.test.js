import React from "react";
import { shallow } from "enzyme";
import { FormComponent } from "../../components/FormComponent/FormComponent";
const formData = [
  {
    fields: [
      {
        name: "id",
        type: "string",
        value: "local1",
      },
      {
        name: "key",
        type: "string",
        value: "A",
      },
      {
        name: "hostname",
        type: "string",
        value: "loalhost1",
      },
      {
        name: "port",
        type: "number",
        value: "8091",
      },
    ],
  },
];
test("Should Render FormComponent", () => {
  const wrapper = shallow(<FormComponent pageData={formData}/>);
  expect(wrapper).toMatchSnapshot();
});
