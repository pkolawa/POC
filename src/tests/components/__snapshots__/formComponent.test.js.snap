// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render FormComponent 1`] = `
<WithStyles(MqFormRow)
  gridSpacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="0"
  >
    <WithStyles(ForwardRef(TextField))
      label="id"
      onChange={[Function]}
      value="local1"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1"
  >
    <WithStyles(ForwardRef(TextField))
      label="key"
      onChange={[Function]}
      value="A"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="2"
  >
    <WithStyles(ForwardRef(TextField))
      label="hostname"
      onChange={[Function]}
      value="loalhost1"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="3"
  >
    <WithStyles(ForwardRef(TextField))
      label="port"
      shrink={true}
      type="number"
      value="8091"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(MqFormRow)>
`;
